gitlab/v4/cli.py

```diff
@@ -53,7 +54,15 @@ def __init__(
         # the class _path attribute, and replace the value with the result.
         if TYPE_CHECKING:
             assert self.mgr_cls._path is not None
-        self.mgr_cls._path = self.mgr_cls._path.format(**self.args)
+        # Items in the path need to be url-encoded
+        if self.mgr_cls._from_parent_attrs:
+            for k in self.mgr_cls._from_parent_attrs:
+                if k in self.args:
+                    self.parent_args[k] = gitlab.utils.clean_str_id(self.args[k])
+                    # If we don't delete it then it will be added to the URL as a
+                    # query-string
+                    del self.args[k]
+        self.mgr_cls._path = self.mgr_cls._path.format(**self.parent_args)
```

Comment:
Cool! Maybe we could extract this into a helper method to keep `__init__` minimal. Then the comments can be a docstring instead.

Reply:
Done